enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java

```diff
@@ -214,4 +259,24 @@ public RoleResult( String role, Set<String> users )
             this.users.addAll( users );
         }
     }
+
+    public class TransactionResult
+    {
+        public final String username;
+        public final String transaction;
+
+        public TransactionResult( KernelTransaction tx )
+        {
+            if ( tx.mode() instanceof AuthSubject )
+            {
+                AuthSubject authSubject = (AuthSubject) tx.mode();
+                username = authSubject.name();
+            }
+            else
+            {
+                username = "-";
+            }
+            transaction = tx.toString();
```

Comment:
Transaction id exists only after commit, it is created here: https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/impl/api/TransactionCommitProcess.java#L48. So it seems not useful for such procedures.

We can print more things about the transaction like `startTimeMillis`, `lastTransactionIdWhenStarted` or `accessMode` but I'm not sure it is useful either.